framework/src/positions/Positions.C

```diff
@@ -50,6 +50,12 @@ Positions::Positions(const InputParameters & parameters)
     _need_sort(getParam<bool>("auto_sort")),
     _initialized(false)
 {
+
+  if (_initial_positions)
+  {
+    auto initial_positions_copy = *_initial_positions;
+    _initial_positions_kd_tree = std::make_unique<KDTree>(initial_positions_copy, 1);
```

Comment:
this is probably a terrible idea.
why does the KDTree not accept a const reference? I ll try to fix that

Reply:
done